ocs_ci/ocs/couchbase.py

```diff
@@ -262,22 +264,42 @@ def get_couchbase_nodes(self):
             nodes_set.add(pod.get().get("spec").get("nodeName"))
         return list(nodes_set)
 
-    def teardown(self):
+    def cleanup(self):
         """
         Cleaning up the resources created during Couchbase deployment
 
         """
+        switch_to_project(constants.COUCHBASE_OPERATOR)
         if self.cb_create_cb_secret:
+            self.cb_secrets._is_deleted = False
             self.cb_secrets.delete()
         if self.cb_create_cb_cluster:
+            self.cb_example._is_deleted = False
             self.cb_example.delete()
         if self.cb_create_bucket:
+            self.cb_bucket._is_deleted = False
             self.cb_bucket.delete()
-        self.subscription_yaml.delete()
+        if self.cb_subscription:
+            self.subscription_yaml._is_deleted = False
+            self.subscription_yaml.delete()
```

Comment:
are we validation either it's delete operation is successful?

Reply:
I keep creating and deleting the individual applications pods for multiple iterations.
_is_deleted flag is set to True for the first iteration [here](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/ocs.py#L138) so the clean up code is getting skipped in the second iteration as the _is_deleted is already set to True. So setting _is_deleted flag to false before deleting the resource.